### PR TITLE
feat: improvement to app (close #116 too)

### LINKED ISSUE #116: Revolt desktop on macos doesn't quit
Neither the quit option in the menu nor the command-Q shortcut allows revolt-desktop to be properly shut down on macos (here on macos Monterey running on a macbook air m1) and revolt has to be force quit.

Tried the two last versions or the desktop application : 1.0.5 and 1.0.6. Same thing.

### PR DESCRIPTION
**Pull request by xhayper**
## Please make sure to check the following tasks before opening and submitting a PR

* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects

- clicking the icon in the dock will unhide it when hidden, and focus it when it is not hidden.
- native arm build for macOS, Windows and Linux.
- force quit (rick click icon and quit or <kbd>⌘</kbd> + <kbd>Q</kbd>) will now actually quit the software.
- added <kbd>Ctrl</kbd> + <kbd>-</kbd> to zoom out.
- make DMG file have background via electron-packer to reduce having to run post script.

close #116

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,15 +77,6 @@
                         "armv7l",
                         "arm64"
                     ]
-                },
-                {
-                    "target": "pacman",
```
Comment (Rexogamer) [inline package.json]: I noticed you removed the Pacman build - how come?

Comment (xhayper) [reply package.json]: It's not the arch's pacman build afaik... so i removed it since i don't know what it was built for.

Comment (xhayper) [reply package.json]: i added the pacman build in the linux fix commit, but figured out soon after that it wasn't arch linux's pacman build so i deleted it.

**Review by insertish — APPROVED** [review]

**Comment by wait-what** [comment]
This PR broke aur/revolt-desktop-git because it removed revolt-desktop.desktop

**Pull request merged**